### ISSUE TITLE
Send a DeadLetter when the actor is stopping 

### ISSUE THREAD

**Issue by alexeyzimarev** [AsynkronIT/protoactor-dotnet#445]
When the actor is gone and there is a message for it,  the `DeadLetter` is published. But when the actor is stopped but not removed from the registry yet, we log an error and then do nothing, so the message gets lost. I believe in such case we should publish a `DeadLetter` as well, thoughts?

```
        public Task InvokeUserMessageAsync(object msg)
        {
            if (_state == ContextState.Stopped)
            {
                //already stopped
                Logger.LogError("Actor {0} already stopped, ignore user message {1} {@2}", Self, msg.GetType().Name, msg);
                return Done;
            }
```

**Comment by potterdai** [comment]
@alexeyzimarev Hmmm, makes sense, it means we should change the code you mentioned to the following?
```
        public Task InvokeUserMessageAsync(object msg)
        {
            if (_state == ContextState.Stopped)
            {
                //already stopped
                DeadLetterProcess.Instance.SendUserMessage(this.Self, msg);
                return Done;
            }
```
A PR is very welcomed = )

**Comment by alexeyzimarev** [comment]
Yes, I did it in my fork. Is delivers more consistent behaviour, because when the actor has been already stopped, the messages to it will be published as dead letter. I see no reason why for a stopping actor messages just vanish.

**Comment by potterdai** [comment]
Fixed in #448
